api/middleware/apd.js

```diff
@@ -3,116 +3,120 @@ const { cache } = require('./cache');
 
 const { getAPDByID: ga } = require('../db');
 
-/**
- * @description Middleware to load an APD into the request "meta" property
- * @param {Object} injection Dependency injection
- * @param {Object} injection.db A Knex object to use for queries
- */
-module.exports.loadApd = ({ getAPDByID = ga } = {}) =>
-  cache(['loadApd'], () => {
-    const loadApd = async (req, res, next) => {
-      logger.silly({ id: req.id, message: 'loading APD for request' });
-      try {
-        const apdFromDB = await getAPDByID(+req.params.id);
+const toCacheLoadApd = ({ getAPDByID = ga } = {}) => async (req, res, next) => {
+  logger.silly({ id: req.id, message: 'loading APD for request' });
+  try {
+    const apdFromDB = await getAPDByID(req.params.id);
+    if (apdFromDB) {
+      const {
+        _id,
+        stateId: state,
+        createdAt: created,
+        updatedAt: updated,
+        status,
+        ...rest
+      } = apdFromDB;
+      req.meta.apd = {
+        ...rest,
+        id: _id.toString(), // eslint-disable-line no-underscore-dangle
+        state,
+        status,
+        created,
+        updated
+      };
+      next();
+    } else {
+      logger.verbose({
+        id: req.id,
+        message: 'requested object does not exist'
+      });
+      res.status(404).end();
+    }
+  } catch (e) {
+    next(e);
+  }
+};
 
-        if (apdFromDB) {
-          req.meta.apd = {
-            ...apdFromDB.document,
-            id: apdFromDB.id,
-            state: apdFromDB.state_id,
-            status: apdFromDB.status,
-            updated: apdFromDB.updated_at
-          };
-          next();
-        } else {
-          logger.verbose({
-            id: req.id,
-            message: 'requested object does not exist'
-          });
-          res.status(400).end();
-        }
-      } catch (e) {
-        next(e);
-      }
-    };
-    return loadApd;
+const toCacheUserCanAccessAPD = ({ loadApd = toCacheLoadApd } = {}) => (
+  req,
+  res,
+  next
+) => {
+  logger.silly({
+    id: req.id,
+    message: 'verifying the user can access this APD'
   });
 
-/**
- * @description Middleware to determine if the current user has access
- *    to the APD they would load.  Also calls loadApd middleware
- * @param {Object} injection Dependency injection
- * @param {Object} injection.loadApd A function that sets req.meta.apd if the
- *    APD exists and then calls a callback. Otherwise terminates the response
- *    and does NOT callback.
- */
-module.exports.userCanAccessAPD = ({ loadApd = module.exports.loadApd } = {}) =>
-  cache(['userCanAccessAPD'], () => {
-    const userCanAccessAPD = async (req, res, next) => {
-      logger.silly({
+  // Load the APD first
+  loadApd()(req, res, () => {
+    // check that the user can view the APD
+    if (
+      req &&
```

Comment:
yeah, so I started to use req?.meta?.state, but that's something that requires babel (I think) and then it complains about like everything else on the backend because it's just straight javascript, maybe I was updating the wrong thing, please tell me if I was.